File: src/Python/easy/string/problem1221.py


class Solution:
    def balancedStringSplit(self, s: str) -> int:
        r = 0
        l = 0
        count = 0
        for idx, char in enumerate(s):
            if char == "R":
                r += 1
            elif char == "L":
                l += 1
            if r == l:
                s = s[idx + 1:]
                count += 1
        return count
  
  
s = "RLRRLLRLRL"
# s = "RLRRRLLRLL"  
# s = "LLLLRRRR"
    
solution = Solution()
solution.balancedStringSplit(s)

File: src/JavaScript/easy/string/problem1221.js
/**
 * @param {string} s
 * @return {number}
 */

var balancedStringSplit = function(s) {
    let r = 0
    let l = 0
    let count = 0
    for(let i = 0; i < s.length; i++){
        if(s[i] == "R"){
            r++
        } else if(s[i] == "L"){
            l++
        }
        if(r == l){
            count++
            s = s.slice(s[i] + 1, s.length)
        }
    }
    return count
};


// let s = "RLRRLLRLRL"
// let s = "RLRRRLLRLL"
let s = "LLLLRRRR"

balancedStringSplit(s)